quickannotator/api/v1/tile/helper.py

```diff
@@ -191,65 +192,40 @@ def generate_random_circle_within_bbox(bbox: Polygon, radius: float) -> shapely.
-        try:
-            # Enable the Spatialite extension on the raw SQLite connection
-            if hasattr(dbapi_connection, "enable_load_extension"):
-                dbapi_connection.enable_load_extension(True)
-                dbapi_connection.execute('SELECT load_extension("mod_spatialite")')
-                dbapi_connection.execute("PRAGMA busy_timeout=10000000;")
-                print("Spatialite extension loaded successfully")
-            else:
-                print("Extension loading not supported for this connection")
-        except Exception as e:
-            print(f"Error enabling Spatialite extension: {e}")
-
-    try:
         # Start a session for the task
-        with Session() as session:
-            # Example: load the tile and process
-            # breakpoint()
-            tile = get_tile(session, annotation_class_id, image_id, tile_id)  # Replace with your actual function to get the tile
-            annotation_class: qadb.AnnotationClass = tile.annotation_class
-            image: qadb.Image = tile.image
-            image_id: int = tile.image_id
-            annotation_class_id: int = tile.annotation_class_id
-
-            # Process the tile (using shapely for example)
-            bbox = get_bbox_for_tile(annotation_class.tilesize, tile_id, image.width, image.height)
-            bbox_polygon = Polygon([(bbox[0], bbox[1]), (bbox[2], bbox[1]), (bbox[2], bbox[3]), (bbox[0], bbox[3])])
-            for _ in range(random.randint(20, 40)):
-                polygon = generate_random_circle_within_bbox(bbox_polygon, 100)
-                insert_new_annotation(session, image_id, annotation_class_id, is_gt=False, tile_id=tile_id, polygon=polygon)
-
-            # Mark tile as processed
-            tile.seen = 2
-            
-            session.commit()
+    with get_session() as db_session:
+        # Example: load the tile and process
+        # breakpoint()
+        tile = get_tile(annotation_class_id, image_id, tile_id)  # Replace with your actual function to get the tile
+        if tile is None:
+            raise ValueError(f"Tile not found: {tile_id}")
+        annotation_class: models.AnnotationClass = tile.annotation_class
+        image: models.Image = tile.image
+        image_id: int = tile.image_id
+        annotation_class_id: int = tile.annotation_class_id
+
+        # Process the tile (using shapely for example)
+        bbox = get_bbox_for_tile(annotation_class.tilesize, tile_id, image.width, image.height)
+        bbox_polygon = Polygon([(bbox[0], bbox[1]), (bbox[2], bbox[1]), (bbox[2], bbox[3]), (bbox[0], bbox[3])])
+        for _ in range(random.randint(20, 40)):
+            polygon = generate_random_circle_within_bbox(bbox_polygon, 100)
+            insert_new_annotation(image_id, annotation_class_id, is_gt=False, tile_id=tile_id, polygon=polygon)
 
-    except Exception as e:
-        print(f"Error processing tile: {e}")
+        # Mark tile as processed
+        tile.seen = 2
```

Comment:
should use ENUM

Reply:
Are you referring to using the ENUM sqlalchemy type? Or more shallowly to use python's enum to refer to these values in the backend?

Reply:
the latter, something more like a "constants.py" which defines all numerical constants globally throughout the project. do you see what i mean?

Reply:
Yes, done. Can you take a look?

Reply:
_very_ close, but i think we'd be better off using an IntEnum

https://docs.python.org/3/library/enum.html#enum.IntEnum

which would allow us to use:

`TileStatus.UNSEEN`

instead of

`TileStatus.UNSEEN.value`



worth taking a look and familiarizing yourself with the other types:
https://docs.python.org/3/howto/enum.html#derived-enumerations

interestingly - i didn't know these existed 10 minutes ago, but the ".value" set me off on a path of discovery :)

Reply:
Just took a look, nice find. I've gone and updated the code

Reply:
Note that the merge commit was necessary due to some conflicts in imports from the dl component. Will need to update the dl component once this PR is merged.

Reply:
Great, thanks! This is now done.